### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by MarcoIeni**
I started creating this file because I wanted to explain that you can't debug with `println!` because the output will be out of sync with the `write!()` macro, but since we are integrating the bat library this part might change.

I decided to do a pull request anyway with the rest of the file. It contains just general info. I don't know if they might be useful for beginners.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- Run Clippy:
+
+  ```shell
+  cargo clippy
+  ```
+
+- Run all tests:
+
+  ```shell
+  cargo test
```
Comment (dandavison) [inline CONTRIBUTING.md:37]: `make test` runs the full set of tests.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+cargo build
+```
+
+### Useful Commands
+
+- Build and run release version:
+
+  ```shell
+  cargo build --release && cargo run --release
+  ```
```
Comment (dandavison) [inline CONTRIBUTING.md:26]: I'm not sure `cargo run --release` is helpful seeing as it requires stdin or two input file args.

**Review by dandavison — COMMENTED** [review]
Thanks for starting this, I think it's a good idea. It might be nice eventually to include some description of how delta works for developers, to help people understand the codebase. I can take a stab at that.

**Comment by dandavison** [comment]
Thanks @MarcoIeni!

**Comment by piyaphart** [comment]
#CONTRIBUTING.md

**Pull request merged**